scala-thoughts.md

```diff
@@ -0,0 +1,15 @@
+### What's easy to do in Scala? What's not?
+There are built-in functions that make pattern-matching recursion seem extraordinarily simple.
+Well, I would say that it is conceptually dense, like Racket. The amount of thought that each line takes is quite a bit, though I may be biased since this is the first time I've used Scala.
+I would say it is not that difficult to do anything, at least compared to other general purpose programming languages. I feel like it is powerful in what I can accomplish in a few lines of code, but at the same time has the ability to take on any task -- I can definitely notice the Java peeking through. 
+It is probably difficult to do domain-specific tasks without building/importing a library that is made for those tasks.
+
+### What is/are your favorite language design choice(s) that the designers of Scala made? Why?
+My favorite language design choice is to have the type declaration after the variable name.
+It feels more intuitive when I am writing out code, and it is easier for me to modify (and recycle) code. 
+
+### What is/are your least favorite language design choice(s)? Why? And why do you think the designers made that / those choice(s)?
+I very much dislike that the language will guess types if I don't declare them explicitly. I love their choice of declaring vals and vars, safety first! I feel like I will get lazy and skip out on type declarations but I think it is generally easier to read code _with_ type declarations and I don't want the language to let me get lazy.
+
+### What Scala features would you like to learn more about?
+Their documentation is quite dense, and lacking examples! I wish I knew more about the various functions that the different built-in types have since they seem to specific to data manipulations that would be tricky or costly to do by hand.
```

Comment:
I too think examples would be nice.

Many functional languages use  a similar set of iterator/list helper methods, and I think that the others provide much better documentation than Scala on them. This is particularly concerning because even though the methods are similar, Scala is the language that has more - meaning their users have probably not seen some of them before!